src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -119,6 +91,13 @@ private void RegisterActions(CompilationStartAnalysisContext compilationStartCon
 
                 compilationStartContext.RegisterSymbolStartAction(symbolStartContext =>
                 {
+                    if (symbolStartContext.Symbol.GetAttributes().Any(a => a.AttributeClass == _structLayoutAttributeType))
+                    {
+                        // Bail out for types with 'StructLayoutAttribute' as the ordering of the members is critical,
+                        // and removal of unused members might break semantics.
+                        return;
+                    }
```

Comment:
This is just invoked for every *named type* we're called back in for, right?

Reply:
IS there a concern about Mef imported fields/props?  Do people ever import them just to have the side effects of that type being loaded?